src/Views/relation_completer.py

```diff
@@ -0,0 +1,35 @@
+"""
+This module contains a custom completer class for generating completion suggestions
+for relation types in a document.
+
+The RelationCompleter class provides completion suggestions for relation types such
+as "aggregation", "composition", "generalization", and "inheritance" based on the
+context of the input text.
+"""
+
+from prompt_toolkit.completion import Completer, Completion
+
+class RelationCompleter(Completer):
```

Comment:
Can't you accomplish all of this by just putting the text in the nested dictionary?

Reply:
Not to my knowledge because the input order is not (add relation type) it's add relation class1 class2 type